src/tabnineChatWidget/handlers/resolveSymbols.ts

```diff
@@ -1,18 +1,46 @@
 import * as vscode from "vscode";
+import * as path from "path";
+
+export type SymbolInformationResult = vscode.SymbolInformation & {
+  relativePath: string;
+};
 
 export async function resolveSymbols({
   symbol,
 }: {
   symbol: string;
-}): Promise<vscode.SymbolInformation[] | undefined> {
+}): Promise<SymbolInformationResult[] | undefined> {
   const symbols = await vscode.commands.executeCommand<
     vscode.SymbolInformation[]
   >("vscode.executeWorkspaceSymbolProvider", symbol);
-  return symbols?.filter((workspaceSymbol) => {
-    const workspacePath = vscode.workspace.workspaceFolders?.[0].uri.path;
-    return (
-      workspacePath &&
-      workspaceSymbol.location.uri.fsPath.startsWith(workspacePath)
-    );
-  });
+
+  if (!symbols) return undefined;
+
+  const workspacePath = vscode.workspace.workspaceFolders?.[0].uri.path;
+
+  if (!workspacePath) return [];
+
+  return symbols
+    .map((workspaceSymbol) => {
+      const symbolPath = workspaceSymbol.location.uri.fsPath;
+      const relativePath = symbolPath.startsWith(workspacePath)
+        ? path.relative(workspacePath, symbolPath)
+        : undefined;
+      if (!relativePath || isProbablyNotSource(relativePath)) {
+        return undefined;
+      }
+      return {
+        ...workspaceSymbol,
+        relativePath,
+      };
+    })
+    .filter(
+      (workspaceSymbol) => !!workspaceSymbol
+    ) as SymbolInformationResult[];
+}
+
+function isProbablyNotSource(symbolPath: string): boolean {
+  return ["node_modules", "dist", "build", "target", "out"].some((dir) =>
+    symbolPath.startsWith(dir)
```

Comment:
r u  sure that in all cases the past will start with any of this?  i mean the relative path can start  with `/`  or  in java it could be a gazilion levels down

Reply:
So what r u suggesting? To check for contains instead of startswith?
Not sure it's a good idea either, I can imagine users naming some sub-folder "target" for example